src/components/radio-button/radio-button.component.tsx

```diff
@@ -0,0 +1,163 @@
+  /** Aria label for rendered help component */
+  helpAriaLabel?: string;
+}
+
+const RadioButton = React.forwardRef<
+  HTMLInputElement,
+  RadioButtonProps & InternalRadioButtonProps
+>(
+  (
+    {
+      autoFocus,
+      checked,
+      disabled,
+      fieldHelp,
+      fieldHelpInline,
+      id,
+      inline,
+      inputWidth,
+      label,
+      labelHelp,
+      labelAlign,
+      labelSpacing = 1,
+      labelWidth,
+      name,
+      onChange,
+      onBlur,
+      onFocus,
+      value,
+      reverse = false,
+      required,
+      size,
+      error,
+      warning,
+      info,
+      tooltipPosition,
+      "data-component": dataComponent = "radio-button",
+      "data-element": dataElement,
+      "data-role": dataRole,
+      helpAriaLabel,
+      ...props
+    }: RadioButtonProps & InternalRadioButtonProps,
+    ref
+  ) => {
+    const marginProps = filterStyledSystemMarginProps(props);
+    const handleChange = useCallback(
+      (event: React.ChangeEvent<HTMLInputElement>) => {
+        /* istanbul ignore else */
+        if (onChange) {
+          onChange(event);
+        }
+
+        // trigger focus, as Safari doesn't focus radioButtons on click by default
+        event.target.focus();
+      },
+      [onChange]
+    );
+
+    const commonProps = {
+      disabled,
+      fieldHelpInline,
+      inputWidth,
+      labelSpacing,
+      error,
+      warning,
+      info,
+    };
+
+    const inputProps = {
+      ...commonProps,
+      autoFocus,
+      checked,
+      fieldHelp,
+      name,
+      onChange: handleChange,
+      onBlur,
+      onFocus,
+      labelAlign,
+      labelInline: true,
+      labelWidth,
+      label,
+      labelHelp,
+      id,
+      value,
+      type: "radio",
+      /**
+       * Invert the reverse prop, to ensure the FormField component renders the components
+       * in the desired order (other elements which use FormField render their sub-components the
+       * opposite way around by default)
+       */
+      reverse: !reverse,
+      required,
+      inputRef: ref,
+      ...props,
+    };
+
+    invariant(
+      !props.children,
+      "This component is meant to be used as a self-closing tag. " +
+        "You should probably use the label prop instead."
+    );
+
+    return (
+      <TooltipProvider
+        helpAriaLabel={helpAriaLabel}
+        tooltipPosition={tooltipPosition}
+      >
+        <RadioButtonStyle
+          data-component={dataComponent}
+          data-role={dataRole}
+          data-element={dataElement}
+          inline={inline}
+          reverse={reverse}
+          size={size}
+          {...commonProps}
+          {...marginProps}
+        >
+          <CheckableInput {...inputProps}>
+            <RadioButtonSvg />
+          </CheckableInput>
+        </RadioButtonStyle>
+      </TooltipProvider>
+    );
+  }
+);
+
+RadioButton.displayName = "RadioButton";
+
+export { RadioButton as PrivateRadioButton };
```

Comment:
I think this line might be removed and we could use the named export as suggested above

Reply:
actually I don't think that it is needed at all